compiam/rhythm/transcription/__init__.py

```diff
@@ -382,3 +388,303 @@ def get_generators(
+    """
+    bōl or solkattu transcription from audio. Based on model presented in [1]
+    
+    [1] Gupta, S., Srinivasamurthy, A., Kumar, M., Murthy, H., & Serra, X. 
+    (2015, October). Discovery of Syllabic Percussion Patterns in Tabla 
+    Solo Recordings. In Proceedings of the 16th International Society 
+    for Music Information Retrieval Conference (ISMIR 2015) (pp. 385–391). 
+    Malaga, Spain.
+    """
+    def __init__(
+        self, 
+        syllables, 
+        feature_kwargs={
+            'n_mfcc':13, 
+            'win_length':1024, 
+            'hop_length':256
+        }, 
+        model_kwargs={
+            'n_components':7, 
+            'n_mix': 3, 
+            'n_iter':100, 
+            'algorithm': 'viterbi',
+            'params':'mcw'
+        }, 
+        sr=44100):
+        """
+        :param syllables: List of strings representing expected bōl/solkattu syllables OR dict of string:string mappings.
+            If a dict is passed, any subsequent syllable labels passed at training time will be converted as per this mapping. The values of
+            this dict will act as the bōl/solkattu "vocabulary" for this model.
+        :type syllables: list or dict
+        :param feature_kwargs: Dict of parameters to pass to librosa.feature.mfcc or librosa.feature.delta. Defaults are selected as per [1]
+        :type feature_kwargs: dict
+        :param model_kwargs: Dict of parameters to pass to hmmlearn.hmm.GMMHMM. Defaults are selected as per [1]
+        :type model_kwargs: dict
+        :param sr: sampling rate of audio to train on (default 44.1Hz)
+        :type sr: int
+        
+        [1] Gupta, S., Srinivasamurthy, A., Kumar, M., Murthy, H., & Serra, X. 
+        (2015, October). Discovery of Syllabic Percussion Patterns in Tabla 
+        Solo Recordings. In Proceedings of the 16th International Society 
+        for Music Information Retrieval Conference (ISMIR 2015) (pp. 385–391). 
+        Malaga, Spain.
+        """ 
+        self.sr = sr
+
+        if isinstance(syllables, dict):
+            self.syllables = set(syllables.values())
+            self.mapping = syllables
+        else:
+            self.syllables = set(syllables)
+            self.mapping = {x:x for x in syllables}
+
+        self.models = {}
+        for s in self.syllables:
+            self.models[s] = hmm.GMMHMM(**model_kwargs)
+
+        self.feature_kwargs = feature_kwargs
+        self.trained = False
+
+    def train(self, filepaths_audio, filepaths_annotation, sr=None):
+        """
+        Train one gaussian mixture model hidden markov model for each syllables passed at initialisation 
+        on input audios and annotations passed via <filepaths_audio> and <filepaths_annotation>.
+        Training hyperparameters are configured upon intialisation and can be accessed/changed
+        via self.model_kwargs.
+
+        :param filepaths_audio: List of filepaths to audios to train on
+        :type filepaths_audio: list
```

Comment:
we have not been adding the `type` on the other docstrings. Maybe we should decide what we do and be consistent.

Reply:
I think it is fine not to have it or have it, the documentation accepts both versions. its not so important, in almost all cases the type is obvious from the param description